Objects/codeobject.c

```diff
@@ -25,7 +28,13 @@ notify_code_watchers(PyCodeEvent event, PyCodeObject *co)
             // callback must be non-null if the watcher bit is set
             assert(cb != NULL);
             if (cb(event, co) < 0) {
-                PyErr_WriteUnraisable((PyObject *) co);
+                // Don't risk resurrecting the object if an unraisablehook keeps
+                // a reference; pass a string as context.
+                PyObject *repr = code_repr(co);
```

Comment:
What if code_repr() fails?

Reply:
Updating to account for the possibility that `code_repr`, `func_repr`, or even `PyUnicode_FromFormat` might fail; in this case we can pass in a more generic context message built with `PyUnicode_FromString`. I think the only way `PyUnicode_FromString` can fail (given a literal string known to be of OK size and not have encoding errors) is an OOM condition? I guess in that case the best we can do is just skip the `PyErr_WriteUnraisable`, given we have no context object we can safely pass to it (and if we are out of memory it will probably fail anyway.)

Reply:
I'm not sure even the generic fallback message via `PyUnicode_FromString` is worth it, given how rare it seems it should be even for `code_repr` or `func_repr` to fail. We could also just skip `PyErr_WriteUnraisable` given a repr failure. But I'll push the version that handles the repr failure a bit more gracefully, and see what you think.

Reply:
Agree that these are very unlikely failure scenarios, but I feel we should at least try to give a useful error. "Errors should never pass silently" :)

Reply:
I decided the least-silent option is to fall back to `Py_None` as context object if we can't even build a useful context string, and still try to call `PyErr_WriteUnraisable` so whatever error happened doesn't pass totally silently.